src/main/java/frc/robot/subsystems/TurretSwivelSubsystem.java

```diff
@@ -5,57 +5,99 @@
 package frc.robot.subsystems;
 
 import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
 import com.ctre.phoenix.motorcontrol.can.TalonSRX;
 
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants;
 import frc.robot.RobotMap;
-//import frc.util.TurretCalibration;
+import frc.util.TurretCalibration;
 
 public class TurretSwivelSubsystem extends SubsystemBase {
 
-    private TalonSRX turretMotor;
-    
+    private TalonSRX _turretMotor;
+    private TurretCalibration _shot;
 
     public TurretSwivelSubsystem() {
-        turretMotor = new TalonSRX(RobotMap.TURRET_MOTOR);
-        //setShot(Constants.TURRET_PID);
+        _turretMotor = new TalonSRX(RobotMap.TURRET_MOTOR);
+
+        _turretMotor.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+
+        setShot(Constants.TURRET_DEFAULT_PID);
     }
 
     @Override
     public void periodic() {
-        // This method will be called once per scheduler run
+        SmartDashboard.putNumber("Turret Angle", getAngle());
+        SmartDashboard.putNumber("Turret RAW Encoder", _turretMotor.getSelectedSensorPosition());
+        SmartDashboard.putNumber("Turret Target", _shot.target);
+        SmartDashboard.putString("Turret PID", _shot.name);
     }
 
     @Override
     public void simulationPeriodic() {
-        // This method will be called once per scheduler run during simulation
+        
     }
 
     public void defaultCommand() {
         
     }
 
-    public void holdTarget(int relativeChange) {
-        
+    /**
+     * Adjusts the heading by the input angle, for holding focus on a target
+     * @param relativeChange Angle of which to add to the current angle.
+     */
+    public void holdTarget(double relativeChange) {
+        double changedAngle = getAngle() + relativeChange;
+        goToAngle(changedAngle);
     }
 
-    public void flip() {
+    public double getAngle() {
+        return _turretMotor.getSelectedSensorPosition() / Constants.TURRET_ENCODER_RATIO;
+    }
 
+    public void flip() {
+        goToAngle(-1 * getAngle());
     }
 
     public void seek() {
-        
+        if(_shot.target == Constants.TURRET_RANGE && getIsAtTarget()) {
+            goToAngle(-1 * Constants.TURRET_RANGE);
+        } else if(getIsAtTarget()) {
+            goToAngle(Constants.TURRET_RANGE);
+        }
     }
 
-    public void goToAngle(int angle) {
-        this.turretMotor.set(ControlMode.Position, angle);
+    public void goToAngle(double angle) {
```

Comment:
Some of these methods seem like they could be private.

Reply:
Moved one-time-use public method code into the commands that used them